components/google_contacts/actions/update-contact/update-contact.mjs

```diff
@@ -0,0 +1,156 @@
+import common from "../common/base.mjs";
+
+export default {
+  ...common,
+  key: "google_contacts-update-contact",
+  name: "Update Contact",
+  description: "Updates a contact. [See the docs here](https://developers.google.com/people/api/rest/v1/people/updateContact)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    ...common.props,
+    resourceName: {
+      propDefinition: [
+        common.props.googleContacts,
+        "resourceName",
+      ],
+    },
+    updatePersonFields: {
+      propDefinition: [
+        common.props.googleContacts,
+        "updatePersonFields",
+      ],
+    },
+  },
+  async additionalProps() {
+    const props = {};
+    if (this.updatePersonFields.includes("names")) {
+      props.firstName = {
```

Comment:
Are all these `props` required when selected? Or should they be `optional`?

Reply:
Whichever `additionalProps` are returned are all required.